pkg/operator/operator.go

```diff
@@ -49,12 +54,37 @@ type Operator struct {
 }
 
 func NewOperator(ctx context.Context, operator *operator.Operator) (context.Context, *Operator) {
-	machineProvider := machine.NewDefaultProvider(ctx, operator.GetClient())
-	machineDeploymentProvider := machinedeployment.NewDefaultProvider(ctx, operator.GetClient())
+	mgmtCluster, err := buildManagementClusterKubeClient(ctx, operator)
+	if err != nil {
+		log.Fatalf("unable to build management cluster client: %v", err)
+	}
+
+	machineProvider := machine.NewDefaultProvider(ctx, mgmtCluster)
+	machineDeploymentProvider := machinedeployment.NewDefaultProvider(ctx, mgmtCluster)
 
 	return ctx, &Operator{
 		Operator:                  operator,
 		MachineProvider:           machineProvider,
 		MachineDeploymentProvider: machineDeploymentProvider,
 	}
 }
+
+func buildManagementClusterKubeClient(ctx context.Context, operator *operator.Operator) (client.Client, error) {
+	if options.FromContext(ctx).ClusterAPIKubeConfigFile != "" {
+		clusterAPIKubeConfig, err := clientcmd.BuildConfigFromFlags("", options.FromContext(ctx).ClusterAPIKubeConfigFile)
+		if err != nil {
+			return nil, err
+		}
+		mgmtCluster, err := cluster.New(clusterAPIKubeConfig, func(o *cluster.Options) {
+			o.Scheme = operator.GetScheme()
```

Comment:
i'm wondering if this is the correct scheme?

when i run this PR with a mgmt cluster and workload cluster i see this error:

```
{"level":"ERROR","time":"2024-07-16T21:07:08.962Z","logger":"controller","message":"Reconciler error","controller":"nodeclaim.garbagecollection","error":"listing machines, no
 kind is registered for the type v1beta1.MachineList in scheme \"sigs.k8s.io/karpenter/pkg/operator/scheme/scheme.go:29\""} 
```

Reply:
i'm thinking it must be correct because it finds the other capi types, but for some reason MachineList doesn't seem to be working.

Reply:
doing a little more testing around this, i think it is unrelated to the change.

Reply:
ok, figured it out. the patch needed to be rebased.

Reply:
I rebased it, and the local test is ok, thank you.